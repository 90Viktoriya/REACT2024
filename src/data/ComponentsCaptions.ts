export enum ComponentsCaptions {
  SEARCH = 'Search',
  ERROR = 'Catch Error',
  NOTHING_FOUND = 'Sorry, nothing found',
  ERROR_MSG = 'You throw the error',
  ERROR_TITLE = 'OOPS! Something went wrong',
  UNKNOWN_VALUE = 'unknown',
  ERROR_PAGE = 'OOPS! Page not found',
  PREV = 'Prev',
  NEXT = 'Next',
  CLOSE = 'Close'
}
