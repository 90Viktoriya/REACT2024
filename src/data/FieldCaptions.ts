export enum FieldCaptions {
  NAME = 'Name: ',
  GENDER = 'Gender: ',
  YEAR_OF_BIRTH = 'Year of birth: ',
  YEAR_OF_DEATH = 'Year of death: ',
  EPISODES = 'Episodes',
  MOVIES = 'Movies',
  PERFORMERS = 'Performers',
  DATE_OF_BIRTH = 'Date of birth: ',
  PLACE_OF_BIRTH = 'Place of birth',
  TITLE = 'Title: ',
  EPISODE_NUMBER = 'Episode number: ',
  RELEASE_DATE = 'Release date: '
}
