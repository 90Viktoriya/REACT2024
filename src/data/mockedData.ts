export const details = {
  uid: 'test12',
  name: 'Detailed card test',
  gender: '',
  yearOfBirth: 0,
  placeOfBirth: 'test',
  yearOfDeath: 0,
  performers: [{ uid: 'qwe', dateOfBirth: '2024', name: 'sd', placeOfBirth: 'aaa' }],
  episodes: [{ uid: 'asd', title: 'qqq', episodeNumber: 3 }],
  movies: [{ uid: 'zxc', title: 'qqq', usReleaseDate: '102' }]
};

export const page = {
  pageNumber: 1,
  pageSize: 1,
  numberOfElements: 1,
  totalElements: 1,
  totalPages: 1,
  firstPage: true,
  lastPage: false
};

export const characters = [
  {
    uid: 'ts',
    name: 'Character1',
    gender: 'M',
    yearOfBirth: 2024
  },
  {
    uid: 'ts2',
    name: 'aaa2',
    gender: 'F',
    yearOfBirth: 2024
  }
];

export const data = {
  page,
  characters
};
