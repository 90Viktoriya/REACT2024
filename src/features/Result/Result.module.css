.main {
  margin: 1em 10em;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
}
.card {
  display: flex;
  flex-direction: column;
  flex-grow: 1;
  flex-basis: 0;
  border: 2px outset var(--dark);
  background-color: var(--light-transparent);
  color: var(--dark);
  padding: 0.2em;
  padding-bottom: 1em;
  margin: 0.2em;
  min-width: 250px;
  max-width: 300px;
  min-height: 150px;
  justify-content: space-between;
}
.title {
  font-size: 2em;
  color: var(--dark);
}
.description {
  color: var(--purple);
}
.nothing {
  color: var(--dark);
}
.span {
  color: var(--dark);
}
@media all and (max-width: 1000px) {
  .main{
    margin: 1em 5em;
  }
}

@media all and (max-width: 700px) {
  .main{
    margin: 1em 1em;
  }
}