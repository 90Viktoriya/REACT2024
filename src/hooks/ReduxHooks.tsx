import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux';
import type { RootState, AppDispatch } from '../store/storeRedux';

export const useAppDispatch = useDispatch.withTypes<AppDispatch>();
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;
//export const useAppSelector: RootState = useSelector.withTypes<RootState>();
