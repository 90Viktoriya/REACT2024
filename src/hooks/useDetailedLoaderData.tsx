import { useLoaderData } from 'react-router-dom';
import { detailedLoader } from '../features/Router/Loaders/detailedLoader';

export const useDetailedLoaderData = () => {
  return useLoaderData() as Awaited<ReturnType<typeof detailedLoader>>;
};
