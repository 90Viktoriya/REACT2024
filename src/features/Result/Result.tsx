import styles from './Result.module.css';
import { ComponentsCaptions } from '../../data/ComponentsCaptions';
import { NavLink, useLoaderData } from 'react-router-dom';
import { Characters, Page } from '../../services/api.types';
import { RouterPath } from '../Router/Router.enum';
import { Pagination } from '../Pagination/Pagination';

export function Result() {
  const { characters } = useLoaderData() as { characters: Characters[]; page: Page };

  if (!characters.length) {
    return <h2 className={styles.nothing}>{ComponentsCaptions.NOTHING_FOUND}</h2>;
  }
  return (
    <>
      <section className={styles.result}>
        {characters.map((item: Characters) => (
          <NavLink
            key={item.uid}
            to={`${RouterPath.DETAILS}/${item.uid}`}
            className={({ isActive }: { isActive: boolean }) =>
              isActive ? `${styles.card} ${styles.active}` : `${styles.card}`
            }
          >
            <p className={styles.title}>{item.name}</p>
            <div className={styles.description}>
              <p>
                Gender: <span className={styles.span}>{item.gender ?? ComponentsCaptions.UNKNOWN_VALUE}</span>
              </p>
              <p className={styles.description}>
                Year of birthday:{' '}
                <span className={styles.span}>{item.yearOfBirth ?? ComponentsCaptions.UNKNOWN_VALUE}</span>
              </p>
            </div>
          </NavLink>
        ))}
      </section>
      <Pagination />
    </>
  );
}
