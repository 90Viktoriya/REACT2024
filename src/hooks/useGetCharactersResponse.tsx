import { useGetCharactersByNameQuery } from '../services/apiRTK';
import { useAppSelector } from './ReduxHooks';

export function useGetCharactersResponse() {
  const searchValue = useAppSelector((state) => state.navigation.searchValue);
  const currentPage = useAppSelector((state) => state.navigation.currentPage);
  const { data } = useGetCharactersByNameQuery({ name: searchValue, page: currentPage });
  return data;
}
