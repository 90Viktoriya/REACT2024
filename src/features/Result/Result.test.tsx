import { describe, expect, Mock, vi } from 'vitest';
import { render } from '@testing-library/react';
import { Result } from './Result';
import { ComponentsCaptions } from '../../data/ComponentsCaptions';
import { MemoryRouter } from 'react-router-dom';
import { useGetCharactersResponse } from '../../hooks/useGetCharactersResponse';

vi.mock('../../hooks/useGetCharactersResponse', () => ({
  useGetCharactersResponse: vi.fn()
}));
vi.mock('../Selector/Selector', () => ({
  Selector: () => <div>Selector</div>
}));
vi.mock('../Pagination/Pagination', () => ({
  Pagination: () => <div>Pagination</div>
}));
describe('Given Result component', () => {
  it('when list of character empty, should render appropriate message ', () => {
    (useGetCharactersResponse as Mock).mockReturnValue({ characters: [], page: 0 });
    const screen = render(<Result />);
    expect(screen.findByText(ComponentsCaptions.NOTHING_FOUND)).toBeTruthy();
  });

  it('when list of character not empty, should display data', () => {
    (useGetCharactersResponse as Mock).mockReturnValue({
      characters: [{ uid: 'test', name: 'Result test', gender: '', yearOfBirth: null }],
      page: 0
    });
    const screen = render(
      <MemoryRouter>
        <Result />
      </MemoryRouter>
    );
    expect(screen.findByText('Result test')).toBeTruthy();
  });
});
