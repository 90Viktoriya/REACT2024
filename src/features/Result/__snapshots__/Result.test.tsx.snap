// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Given Result component > when rendered, should match snapshot 1`] = `
<DocumentFragment>
  <h2
    class="_nothing_cc01d9"
  >
    Sorry, nothing found
  </h2>
</DocumentFragment>
`;
